core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaStoreReaderThread.java

```diff
@@ -147,7 +131,7 @@ public void doWork() {
           V message = null;
           try {
             message =
-                messageBytes == null ? null : serializer.deserializeValue(messageKey, messageBytes);
+                record.value() == null ? null : serializer.deserializeValue(messageKey, record.value());
           } catch (SerializationException e) {
             log.error("Failed to deserialize a schema or config update", e);
```

Comment:
Not part of this patch, but it seems like a bug to ignore this exception and continue below. Wouldn't we then incorrectly see the message as an instance of null and delete the key?